frontend/public/components/factory/table.tsx

```diff
@@ -552,320 +386,295 @@ const getActiveColumns = (
-          });
-        });
-      }
+const getComponentProps = (
+  data: any[],
+  filters: Filter[],
+  selected: boolean,
+  match: RMatch<any>,
+  kindObj: K8sResourceKindReference,
+): ComponentProps => ({
+  data,
+  filters,
+  selected,
+  match,
+  kindObj,
+});
 
-      componentDidMount() {
-        const componentProps: ComponentProps = _.pick(this.props, [
-          'data',
-          'filters',
-          'selected',
-          'match',
-          'kindObj',
-        ]);
-        const columns = getActiveColumns(
-          this.props.Header,
-          componentProps,
-          this.props.activeColumns,
-          this.props.columnManagementID,
-          this.props.showNamespaceOverride,
-        );
-        const sp = new URLSearchParams(window.location.search);
-        const columnIndex = _.findIndex(columns, { title: sp.get('sortBy') });
+export const Table: React.FC<TableProps> = ({
+  onSelect,
+  filters,
+  selected,
+  match,
+  kindObj,
+  Header,
+  activeColumns,
+  columnManagementID,
+  showNamespaceOverride,
+  scrollElement,
+  Rows,
+  Row,
+  expand,
+  label,
+  mock,
+  selectedResourcesForKind,
+  'aria-label': ariaLabel,
+  virtualize = true,
+  customData,
+  gridBreakPoint = TableGridBreakpoint.none,
+  loaded,
+  loadError,
+  NoDataEmptyMsg,
+  EmptyMsg,
+  defaultSortOrder,
+  customSorts,
+  data: propData,
+  defaultSortFunc,
+  reduxID,
+  reduxIDs,
+  staticFilters,
+  rowFilters,
+  isPinned,
+  defaultSortField,
+}) => {
+  const { currentSortField, currentSortFunc, currentSortOrder, data, listId } = useTableData({
+    reduxID,
+    reduxIDs,
+    defaultSortFunc,
+    defaultSortField,
+    defaultSortOrder,
+    staticFilters,
+    filters,
+    rowFilters,
+    propData,
+    loaded,
+    isPinned,
+    customData,
+    customSorts,
+    sorts,
+  });
 
-        if (columnIndex > -1) {
-          const sortOrder = sp.get('orderBy') || SortByDirection.asc;
-          const column = columns[columnIndex];
-          this._applySort(column.sortField, column.sortFunc, sortOrder, column.title);
-          this.setState({
-            sortBy: {
-              index: columnIndex + this._columnShift,
-              direction: sortOrder,
-            },
-          });
-        }
+  const dispatch = useDispatch();
+  const { t } = useTranslation();
+  const [windowWidth, setWindowWidth] = React.useState(window.innerWidth);
+  const [sortBy, setSortBy] = React.useState({});
+
+  const columns = React.useMemo(
+    () =>
+      getActiveColumns(
+        windowWidth,
+        Header,
+        getComponentProps(data, filters, selected, match, kindObj),
+        activeColumns,
+        columnManagementID,
+        showNamespaceOverride,
+        t,
+      ),
+    [
+      windowWidth,
+      Header,
+      data,
+      filters,
+      selected,
+      match,
+      kindObj,
+      activeColumns,
+      columnManagementID,
+      showNamespaceOverride,
+      t,
+    ],
```

Comment:
In these dependency lists, can you confirm none of these items are changing every render? I just want to make sure we're not passing anonymous functions or similar as props somewhere causing the body to reevaluate each time.

Reply:
good idea.
`Header` is problematic, many times passed as new function.
`filters` can also cause issues I think.
The other dependencies are fine.

There's the same problem in `table-data-hook` where all filter and sort props can cause unnecessary rerenders. 

I solved it by using `useDeepCompareMemoize` but we will want to fix the parent components so we dont have to use the deep compare in the future.

Reply:
FYI there is no need to use the deep memoize in new table proposal https://github.com/openshift/console/pull/8829